core/src/autogluon/core/scheduler/fifo.py

```diff
@@ -363,7 +367,8 @@ def add_job(self, task, **kwargs):
             self.scheduled_tasks.append(task_dict)
 
     def _clean_task_internal(self, task_dict):
-        task_dict['ReporterThread'].join()
+        while task_dict['ReporterThread'].is_alive():
+            task_dict['ReporterThread'].join(timeout=60)
```

Comment:
maybe add a comment describing what this does and why its needed

Reply:
I don't think this is needed anymore - these are left-overs from earlier work. I'll revert it back